### PR TITLE
Feature/support for latest nvim keymap functions

### PR DESCRIPTION
**Pull request by Issafalcon**

### CODE REVIEW

File: lua/nvim-mapper.lua [hunk 1]
```diff
@@ -3,101 +3,115 @@ local M = {}
...
-    end
-
-    -- Set the mapping
-    if not virtual then
-        if buffnr ~= nil then
-            vim.api.nvim_buf_set_keymap(buffnr, mode, keys, cmd, options)
-        else
-            vim.api.nvim_set_keymap(mode, keys, cmd, options)
-        end
+  -- Set the mapping
```
Comment (Issafalcon) [inline lua/nvim-mapper.lua:38]: Most of this file is just sumneko_lua LSP formatting. This block is the only one that has actual code changes

Comment (Issafalcon) [inline lua/nvim-mapper.lua:38]: Changes in this block

File: lua/telescope/_extensions/mapper/utils.lua [hunk 2]
```diff
@@ -5,119 +5,143 @@ local M = {}
...
+      mode_str = mode_str .. "operator pending"
     elseif (mode == "s") then
-        mode_str = "select"
+      mode_str = mode_str .. "select"
     elseif (mode == "r") then
-        mode_str = "replace"
+      mode_str = mode_str .. "replace"
     else
-        mode_str = mode
+      mode_str = mode_str .. mode
```
Comment (Issafalcon) [inline lua/telescope/_extensions/mapper/utils.lua:122]: Changes here to create string from table of modes

**Comment by Issafalcon** [comment]
Sorry for formatting. I'm not sure what rules you normally use in your .toml file. I've commented on where actual changes are. Feel free to re-format using your preferred rules (or add a stylua.toml file to the repo?)

**Review by lazytanuki — APPROVED** [review]
LGTM, thanks ! :)

**Pull request merged**